pkg/controllers/nodeclass/suite_test.go

```diff
@@ -232,6 +230,30 @@ var _ = Describe("NodeClass Termination", func() {
 		Expect(awsEnv.IAMAPI.InstanceProfiles).To(HaveLen(0))
 		ExpectNotFound(ctx, env.Client, nodeClass)
 	})
+
+	It("should succeed to delete legacy instance profile if one exists and the NodeClass is deleted", func() {
+		profileName := nodeClass.LegacyInstanceProfileName(options.FromContext(ctx).ClusterName, fake.DefaultRegion)
+		awsEnv.IAMAPI.InstanceProfiles = map[string]*iamtypes.InstanceProfile{
+			profileName: {
+				InstanceProfileName: lo.ToPtr(profileName),
+				Roles: []iamtypes.Role{
+					{
+						RoleId:   aws.String(fake.RoleID()),
+						RoleName: aws.String("fake-role"),
+					},
+				},
+			},
+		}
+		controllerutil.AddFinalizer(nodeClass, v1.TerminationFinalizer)
+		ExpectApplied(ctx, env.Client, nodeClass)
+		ExpectObjectReconciled(ctx, env.Client, controller, nodeClass)
```

Comment:
Any reason to reconcile here and test that we create a new instance profile to replace the legacy profile?